docs: update service-worker-config to more clearly describe behavior

jekru opened the pull request:
Fixes #45280

jekru committed:
docs: update service-worker-config to more clearly describe behavior

Fixes #45280

gkalpak commented on aio/content/guide/service-worker-config.md:
```diff
@@ -222,6 +224,14 @@ Only non-mutating requests \(GET and HEAD\) are cached.
+
+**Example** `/foo/bar/*.js` will match any .js file but also any .json file. 
+
+* `^` is another character with a special meaning in a regular expression and matches the position before the first character in a string. This character will not be added in the conversion from glob to regex and it is recommended to add `^` yourself at the beginning of the url(s).
```
I wouldn't say it is recommended (because some URLs are just partial patterns). Let's just say that if you want your pattern to match the beginning of the URL, you should add it :smiley:

gkalpak reviewed (changes requested):
Thx, @jekru, and sorry it took me so long to get around to it 😇
I left a couple of suggestions.

gkalpak commented on aio/content/guide/service-worker-config.md:
```diff
@@ -30,7 +30,21 @@ Unless otherwise commented, patterns use a **limited*** glob format that interna
+
+*   `$` is a special character in regex that matches the end of a line and will not be escaped in the internal conversion from glob to regex format. Be aware that you have to escape it yourself, otherwise your url can result in an unmatchable regular expression.
+
+<div class="alert is-helpful">
+`foo/bar/$value` is an unmatchable expression because it is impossible to have a string that has any characters after it has ended.
+</div>
```
Formatting suggestion:

```suggestion
    <div class="alert is-helpful">

      `foo/bar/$value` is an unmatchable expression because it is impossible to have a string that has any characters after it has ended.

    </div>
```

(Note that for our Markdown parser to work correctly, there should be a empty line when switching from HTML to Markdown.)

Same for the other alert below.

gkalpak commented on aio/content/guide/service-worker-config.md:
```diff
@@ -30,7 +30,21 @@ Unless otherwise commented, patterns use a **limited*** glob format that interna
-**\*** Pay attention attention that some characters with a special meaning in a regular expression are not escaped (or added) in the internal glob to regex conversion and can result in unexpected behavior.
+**\*** Pay attention that some characters with a special meaning in a regular expression are not escaped and also the pattern is not wrapped in `^`/`$` in the internal glob to regex conversion and can result in unexpected behavior.
+
+*   `$` is a special character in regex that matches the end of a line and will not be escaped in the internal conversion from glob to regex format. Be aware that you have to escape it yourself, otherwise your url can result in an unmatchable regular expression.
```
Please start each sentence on a new line (here and below):

```suggestion
*   `$` is a special character in regex that matches the end of a line and will not be escaped in the internal conversion from glob to regex format.
    Be aware that you have to escape it yourself, otherwise your url can result in an unmatchable regular expression.
```

gkalpak commented on aio/content/guide/service-worker-config.md:
```diff
@@ -30,7 +30,21 @@ Unless otherwise commented, patterns use a **limited*** glob format that interna
+`/foo/bar/*.js` will match any .js file but also any .json file.
+</div>
+
+* `^` is another character with a special meaning in a regular expression and matches the position before the first character in a string. This character will not be added in the conversion from glob to regex. If you want your pattern to match the beginning of the url(s), you should add it.
```
Can you please uppercase `URL` (to maintain consistency throughout the doc) :pray:

gkalpak reviewed (changes requested):
Thx for making the changes, @jekru.
Left a few more comments (mostly formatting).

gkalpak committed:
fixup! docs: update service-worker-config to more clearly describe behavior

gkalpak reviewed (approved):
Thx again for making the changes, @jekru :+1:

I rebased this on latest `main` and squashed the commits to get CI green.
As I was looking at the final text, I realized the whole section would fit better into the guide flow it is were an alert, so I went ahead and made some adjustment. I added them in a separate fixup commit, so my changes are easier to review.

Can you please take a look and lmk what you think? If the changes look good to you, we can go ahead and get this merged 🚀 

Thx again for working on this and following through with my review suggestions :heart:

jekru commented:
> Thx again for making the changes, @jekru 👍
> 
> I rebased this on latest `main` and squashed the commits to get CI green. As I was looking at the final text, I realized the whole section would fit better into the guide flow it is were an alert, so I went ahead and made some adjustment. I added them in a separate fixup commit, so my changes are easier to review.
> 
> Can you please take a look and lmk what you think? If the changes look good to you, we can go ahead and get this merged 🚀
> 
> Thx again for working on this and following through with my review suggestions ❤️

Looks good to me!

alxhub commented:
This PR was merged into the repository by commit e235c70d7d38c50561f14b08ce17021e1d357c69.
